tests/unit/test_locationcodes.py

```diff
@@ -0,0 +1,77 @@
+"""Tests for strtools.py"""
+
+import time
+import timeit
+import importlib
+
+import pytest
+from scipy import stats
+
+import zolltools.nemsis.locationcodes as locationcodes
+
+@pytest.mark.slow
+def test_get_mapping_performance():
+    """
+    measuring performance of get_mapping
+    
+    Takes a sample of the execution time for the first and successive calls to
+    `get_mapping` and determines that successive calls are at least
+    `min_exp_speedup` times faster than the first calls within a certain
+    confidence interval (see `alpha` and comparison to p-value in `assert`
+    statement).
+    """
+
+    alpha = 0.05             # confidence threshold
+    min_exp_speedup = 40     # successive reads should be min_exp_speedup times faster
+    num_data_points = 10_000
+    data = {"successive": [], "adjusted": []}
+    for _ in range(num_data_points):
+        importlib.reload(locationcodes)
+        timeit_repeat_num = 100
+        seconds_to_ns = 1e9
+
+        # Record for the first read of the mapping
```

Comment:
This is oversampling successive reads as compared to first reads. Can we select a random read instead of the average of `timeit_repeat_num` reads?

Reply:
resolved by new changes